app/services/check_availability_task_service.rb

```diff
@@ -2,6 +2,8 @@ class CheckAvailabilityTaskService < TaskService
   attr_reader :task
 
   def process
+    raise "Source #{source_id} does not exist" if @source.blank?
```

Comment:
@hsong-rh Shouldn't this be 
raise "Source #{source_id} does not exist" unless @source